Uniswap/v4-core
# Enhancements and fixes for SafeCast library and tests

This PR introduces the following changes:
- Fixed issues in the SafeCast library.
- Added comprehensive test cases for SafeCast conversions.
- All tests are passing, ensuring the stability and accuracy of the library.

Steps to reproduce and validate:
- Run `forge test --match-contract SafeCastTest` to verify the test suite.

Looking forward to your feedback and suggestions.
